bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/capability/WeatherCapability.java

```diff
@@ -42,7 +39,7 @@ protected List<NAObject> getFreshData(WeatherApi api) {
         try {
             return List.of(owned ? api.getOwnedStationData(handler.getId()) : api.getStationData(handler.getId()));
         } catch (NetatmoException e) {
-            logger.warn("Error retrieving weather data '{}' : {}", handler.getId(), e.getMessage());
+            statusReason = "Error retrieving weather data '%s' : %s".formatted(handler.getId(), e.getMessage());
```

Comment:
I did a test simulating HTTP status code 503 which causes an exception to be thrown. However, nothing is visible in logs and the Thing status is not updated. It seems related to this place, because this is where the exception is caught. I don't know of the Thing status would update prior to the changes in this PR.

The only thing logged is the HTTP code and response payload:
```
2024-03-23 16:59:09.101 [TRACE] [mo.internal.handler.ApiBridgeHandler] -  -returned: code [503 Service Unavailable] body [...]
```

Reply:
I don't fully understand the design, but it seems like `statusReason` should be set during `Capability.setNewData` as it will be returned from this method and used for setting Thing status.

I'm not sure, but it seems like setting `statusReason` on this path will have no effect. If set in `updateNAThing` it would have effect, since called by `setNewData`, but not in `freeJobAndReschedule`:

https://github.com/openhab/openhab-addons/blob/cac9e2ccda5fc2ab594b0a8cf1560c32260b6a5a/bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/capability/RefreshCapability.java#L100-L124

freeJobAndReschedule → proceedWithUpdate → CommonInterface.proceedWithUpdate → updateReadings → RestCapability.updateReadings → CacheWeatherCapability.updateReadings → getFreshData

Reply:
You're correct, it was a falsely good option, this can't work.

Reply:
I'm resolving this comment, since the change was reverted.